src/Polly.Shared/Duration/DecorrelatedJitterBackoff.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Polly.Duration
+{
+    /// <summary>
+    /// Generates sleep durations in an jittered manner, making sure to mitigate any correlations.
+    /// For example: 1s, 3s, 4s, 2s...
+    /// For background, see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/.
+    /// </summary>
+    public sealed class DecorrelatedJitterBackoff : ISleepDurationStrategy
+    {
+        private readonly ConcurrentRandom _random;
+
+        /// <summary>
+        /// Whether the first retry will be immediate or not.
+        /// </summary>
+        public bool FastFirst { get; }
+
+        /// <summary>
+        /// The minimum duration value to use for each retry.
+        /// </summary>
+        public TimeSpan MinDelay { get; }
+
+        /// <summary>
+        /// The maximum duration value to use for each retry.
+        /// </summary>
+        public TimeSpan MaxDelay { get; }
+
+        /// <summary>
+        /// Creates a new instance of the class.
+        /// </summary>
+        /// <param name="minDelay">The minimum duration value to use for each retry.</param>
+        /// <param name="maxDelay">The maximum duration value to use for each retry.</param>
+        /// <param name="fastFirst">Whether the first retry will be immediate or not.</param>
+        /// <param name="random">An optional <see cref="Random"/> instance to use.
+        /// If not specified, will use a shared (singleton) instance with a random seed.</param>
+        public DecorrelatedJitterBackoff(TimeSpan minDelay, TimeSpan maxDelay, bool fastFirst = false, Random random = null)
```

Comment:
The last param `random` is there for determinism and/or unit testing. You may want a specific seed or you may want a repeatable sequence (as you often do in units)